gpii/node_modules/contextManager/src/ContextManagerUtilities.js

```
@@ -0,0 +1,101 @@
+});
+
+
+gpii.contextManager.transforms.timeInRange = function (value, transformSpec) {
+    var timeInRange = gpii.contextManager.transforms.timeInRange;
+    var currentTime = timeInRange.timeParser(value);
+    var fromTime = timeInRange.timeParser(transformSpec.from);
+    var toTime = timeInRange.timeParser(transformSpec.to);
+
+    if (timeInRange.isEarlier(toTime, fromTime)) { //if time wraps
+        return (timeInRange.isEarlier(fromTime, currentTime) || timeInRange.isEarlier(currentTime, toTime));
+    } else {
+        return (timeInRange.isEarlier(fromTime, currentTime) && timeInRange.isEarlier(currentTime, toTime));
+    }
+};
+
+gpii.contextManager.transforms.timeInRange.timeParser = function (time) {
+    time = time.split(":");
+    return {
+        hours: parseInt(time[0], 10),
+        minutes: parseInt(time[1], 10)
+    };
+};
+
+// checks whether 'first' is earlier than 'second' - returns true if so
+gpii.contextManager.transforms.timeInRange.isEarlier = function (first, second) {
+    return (60 * (first.hours - second.hours) + first.minutes - second.minutes) < 0;
+};
+
+gpii.contextManager.utils.findActiveContexts = function (currentContext, matchData) {
+    var activeContexts = [ "gpii-default" ];
+    // do nothing if there is no NP set yet
+    fluid.each(matchData.inferredConfiguration, function (content, contextId) {
+        if (content.conditions === undefined) {
+            return;
+        }
+        // massage conditions to be proper transform specs:
+        var condArray = [];
+        fluid.each(content.conditions, function (condition) {
+            var entry = fluid.copy(condition);
+            entry.type = gpii.contextManager.utils.transformTypeLookup[condition.type];
+            if (entry.type === undefined) {
+                console.log("ERROR: Failed to find the condition type: " + condition.type + " - Ignoring that conditional block");
+                condArray = false;
+                return;
```

Use fluid.fail rather than early return and strange falsy signalling value.